mac/FreeChat/Models/NPC/LlamaServer.swift

```diff
@@ -269,30 +262,25 @@ actor LlamaServer {
     interrupted = false
     serverErrorMessage = ""
 
-    outputPipe.fileHandleForReading.readabilityHandler = { handle in
-      let data = handle.availableData
-      let lines = String(decoding: data, as: UTF8.self)
-
-      #if DEBUG
-        print(lines)
-      #endif
-
-      Task {
-        await self.handleServerOutput(lines)
-      }
-    }
-
     guard let modelPath = self.modelPath else { return }
     let modelName =
       modelPath.split(separator: "/").last?.map { String($0) }.joined() ?? "Unknown model name"
 
+    let serverHealth = ServerHealth()
+    await serverHealth.updateURL(url("/health"))
+    await serverHealth.check()
+
     var timeout = 60
     let tick = 1
     while true {
-      if serverUp || interrupted { break }
+      await serverHealth.check()
+      let score = await serverHealth.score
+      if score >= 0.25 { break }
+      await serverHealth.check()
       if !process.isRunning {
```

Comment:
Since this is the local server, maybe it should have a separate monitor only for the process, and outside of the start method; maybe the server can have a restart button. The HTTP health check can be in the UI somewhere, to indicate about problems.

Reply:
yeah, separate process makes sense but I don't think I want to expose any UI or have users think about it. The app just restarts it if it goes down. The server local host setup is more of an implementation detail and could be switched out for an in-process version (like in the SwiftUI llama.cpp example) if that performs better. I had set it up this way before that example existed. 

Thanks for taking a look and providing the info and feedback!